django/db/backends/base/schema.py

```diff
@@ -433,6 +434,21 @@ def add_field(self, model, field):
         db_params = field.db_parameters(connection=self.connection)
         if db_params['check']:
             definition += " " + self.sql_check_constraint % db_params
+        if field.remote_field and self.connection.features.supports_foreign_keys and field.db_constraint:
+            constraint_suffix = "_fk_%(to_table)s_%(to_column)s"
+            # Add FK constraint inline, if possible
+            if self.sql_add_field_inline_fk:
+                to_table = field.remote_field.model._meta.db_table
+                to_column = field.remote_field.model._meta.get_field(field.remote_field.field_name).column
+                definition += " " + self.sql_add_field_inline_fk % {
+                    "name": self._fk_constraint_name(model, field, constraint_suffix),
+                    "column": self.quote_name(field.column),
+                    "to_table": self.quote_name(to_table),
+                    "to_column": self.quote_name(to_column),
+                }
+            # Otherwise, add FK constraints later
+            else:
+                self.deferred_sql.append(self._create_fk_sql(model, field, constraint_suffix))
```

Comment:
I don't _think_ it should matter? My understanding of how this works is that this is simply appending to a list of SQL statements which get executed [when the context manager exits](https://github.com/django/django/blob/master/django/db/backends/base/schema.py#L109-L110)—and only then if no exception were raised.

It *could* have made a difference if it weren't for the exception check, as previously an exception being raised in `execute` would prevent the deferred SQL statement from even being appended to the list. But given that [it won't be executed anyway if there's an exception](https://github.com/django/django/blob/master/django/db/backends/base/schema.py#L108), I believe there is functionally no difference here.

Anyway... all of the tests pass ;)

All that said, if you'd prefer, I could finesse this code a bit so that the default cause (where `sql_create_column_inline_fk` isn't defined) is truly identical to the way it was before. I personally think it would make the code a bit more complicated than it needs to be, but it isn't exactly the hill I'd choose to die on.